src/System/Win32/Compat.hs

```diff
@@ -110,3 +125,31 @@ foreign import WINDOWS_CCONV unsafe "_get_osfhandle"
 withStablePtr :: a -> (StablePtr a -> IO b) -> IO b
 withStablePtr value = bracket (newStablePtr value) freeStablePtr
 #endif
+
+-- | `withHandleToHANDLE` does not behave as expected for GHC option
+-- -with-rtsopts=--io-manager=native when Win32 < 2.13.2. Taken from
+-- package Win32-2.13.2.0 `System.Win32.Types.withHandleToHANDLENative`.
+#if defined(__IO_MANAGER_WINIO__)&&!MIN_VERSION_Win32(2,13,2)
```

Comment:
Where is he __IO_MANAGER_WINIO__ macro set?

Reply:
It is emitted by GHC but I see now I have misunderstood its function; it is emitted when GHC is ‘WinIO aware’, not only when GHC is ‘WinIO enabled’. I need to import and use `<!>`. I’ll fix.

Reply:
Could you also set it up so that both CPP branches are checked on the CI? (Just to make sure both compile, even though one of them might not work correctly at runtime.)

Reply:
I think I have now updated the CI to do that.

Reply:
Cheers!